packages/@aws-cdk/aws-rds/lib/cluster-engine.ts

```diff
@@ -160,12 +175,14 @@ interface MysqlClusterEngineBaseProps {
   readonly engineVersion?: EngineVersion;
   readonly defaultMajorVersion: string;
   readonly combineImportAndExportRoles?: boolean;
+  readonly supportsLambda?: boolean;
```

Comment:
Changes based off my comments above would likely change this drastically, so I'll leave this logic alone for the moment.

Reply:
One of the difficulties with this is MySQL engines always supports the S3 Integrations, but only sometimes support the Lambda integration. That is why this exists. 

`combineImportAndExportRoles` is similar because some versions cannot have separate roles for S3 import/export.